regparser/tree/reg_text.py

```diff
@@ -84,6 +85,95 @@ def build_subpart(text, part):
+
+        Pe
+        Pe.
+        Pen
+        Pen.
+        Pena
+        Pena.
+        <etc.>
+        Penalties.
+        Penalties-b.
+        Penalties-c.
+        <etc.>
+        """
+        word = words[0]
+        terminators = ("", ".")
+        suffix_pos, terminator_pos = 0, 0
+        pos = min([2, len(word)])
+        while candidate_text == "" or candidate_text in letter_list:
+            suffix = '-%s' % suffixes[suffix_pos] if suffix_pos else ''
+            candidate_text = '%s%s%s' % (word[:pos],
+                                         terminators[terminator_pos], suffix)
+            if not terminator_pos:
+                terminator_pos = 1
+                continue
+            else:
+                terminator_pos = 0
+            if pos < len(word):
+                pos = pos + 1
+            else:
+                suffix_pos = suffix_pos + 1
+        return (candidate_text, letter_list + [candidate_text])
+    else:
+        """
+        E.g. if the title is "Change of Ownership" the progression is:
+
+        CoO
+        C. o. O. 
+        C-o-O
+        C_o_O
+        ChofOw
+        Ch. of. Ow. 
+        <etc.>
+        """
+        separators = ("", ". ", "-", "_")
+        separator_pos, suffix_pos = 0, 0
+        num_letters = 1
+        longest = max([len(word) for word in words])
+        while candidate_text == "" or candidate_text in letter_list:
+            sep = separators[separator_pos]
+            suffix = suffixes[suffix_pos]
+            suffix = "-%s" % suffix if suffix else ""
+            suffix = "%s%s" % (sep, suffix) if sep == ". " else suffix
+            candidate_text = "%s%s" % (sep.join(
+                word[:num_letters] for word in words), suffix)
+            if separator_pos + 1 < len(separators):
+                separator_pos = separator_pos + 1
+                continue
+            elif num_letters == longest:
+                separator_pos = 0
+                suffix_pos = suffix_pos + 1
+            else:
+                separator_pos = 0
+                num_letters = num_letters + 1
+        return (candidate_text, letter_list + [candidate_text])
+
+def build_subjgrp(text, part, letter_list):
+    """
+    We're constructing a fake "letter" here by taking the first letter of
+    each word in the subjgrp's title, or using the first two letters of the
+    first word if there's just one—we're avoiding single letters to make sure we
+    don't duplicate an existing subpart, and we're hoping that the initialisms
+    created by this method are unique for this regulation.
+    We can make this more robust by accepting a list of existing initialisms and
+    returning both that list and the Node, and checking against the list as we
+    construct them.
+    """
+    letter_text, letter_list = subjgrp_label(text, letter_list)
+    label = [str(part), 'Subpart', letter_text]
+
+    return (letter_list, struct.Node(
+        "", [], label, text, node_type=struct.Node.SUBPART))
```

Comment:
The text and children fields have defaults, so this might be cleaner:
`struct.Node(label=label, title=text, node_type=struct.Node.SUBPART)`

It might be a good idea to s/text/title throughout, really